lib/membrane_rtsp_plugin/source.ex

```diff
@@ -114,13 +141,18 @@ defmodule Membrane.RTSP.Source do
       ) do
     case Enum.find(state.tracks, fn track -> track.rtpmap.payload_type == pt end) do
       nil ->
-        {[], state}
+        raise "No track of payload type #{inspect(pt)} has been requested with SETUP"
 
       track ->
-        ssrc_to_track = Map.put(state.ssrc_to_track, ssrc, track)
+        ssrc_to_track = Map.put(state.ssrc_to_track, ssrc, Map.delete(track, :transport))
 
-        {[notify_parent: {:new_track, ssrc, Map.delete(track, :transport)}],
-         %{state | ssrc_to_track: ssrc_to_track}}
+        state = %{state | ssrc_to_track: ssrc_to_track}
+
+        if map_size(ssrc_to_track) == length(state.tracks) do
```

Comment:
Batching all the notifications may not work well with some rtsp sources (especially ip cameras).  An IP camera usually advertise a track of type `:application` where it is streaming onvif and camera specific events. The problem is those events are only sent when triggered or worse sometimes they are not even sent.

So in this case if I specify `:application` track, it may delay or hang forever because an event is not triggered.

Reply:
That is indeed problematic. One of the reason why I did this change is that in one of our use cases we need to know when all tracks have been set up. Since batching these notifications won't work I think the source can just emit another notification about all tracks that have been set up. The parent of the source can then see how many tracks have been set up and how many received by the SessionBin and work with that.